pyo3-0.20.0 on ppc32: no `AtomicI64` in `sync::atomic`

### Bug Description

When building a package (html-py-ever example from setuptools-rust) using pyo3 0.20.0 on ppc32, I'm getting the following error:

```
error[E0432]: unresolved import `std::sync::atomic::AtomicI64`
 --> /home/mgorny/.cargo/registry/src/index.crates.io-9850fa1df3136423/pyo3-0.20.0/src/impl_/pymodule.rs:6:25
  |
6 | use std::sync::atomic::{AtomicI64, Ordering};
  |                         ^^^^^^^^^
  |                         |
  |                         no `AtomicI64` in `sync::atomic`
  |                         help: a similar name exists in the module: `AtomicI32`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `pyo3` (lib) due to previous error
```

### Steps to Reproduce

```
git clone https://github.com/PyO3/setuptools-rust/
cd setuptools-rust/examples/html-py-ever
python -m build -w
```

### Backtrace

```shell
Running `rustc --crate-name pyo3 --edition=2021 /home/mgorny/.cargo/registry/src/index.crates.io-9850fa1df3136423/pyo3-0.20.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=135 --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="default"' --cfg 'feature="extension-module"' --cfg 'feature="indoc"' --cfg 'feature="macros"' --cfg 'feature="pyo3-macros"' --cfg 'feature="unindent"' -C metadata=c3280b3684a9dc16 -C extra-filename=-c3280b3684a9dc16 --out-dir /home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps -L dependency=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps --extern cfg_if=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libcfg_if-dd66dfbd15e1f200.rmeta --extern indoc=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libindoc-330c4043caad1819.so --extern libc=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/liblibc-bd25ee420bfedf7b.rmeta --extern memoffset=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libmemoffset-80b40800efcae77e.rmeta --extern parking_lot=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libparking_lot-166554cbb51c52ff.rmeta --extern pyo3_ffi=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libpyo3_ffi-d8ceef02bf595e0b.rmeta --extern pyo3_macros=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libpyo3_macros-b6c77a4bfb6a873a.so --extern unindent=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libunindent-a0d1a359c22931b4.rmeta --cap-lints allow --cfg Py_3_6 --cfg Py_3_7 --cfg Py_3_8 --cfg Py_3_9 --cfg Py_3_10 --cfg Py_3_11 --cfg thread_local_const_init`
error[E0432]: unresolved import `std::sync::atomic::AtomicI64`
 --> /home/mgorny/.cargo/registry/src/index.crates.io-9850fa1df3136423/pyo3-0.20.0/src/impl_/pymodule.rs:6:25
  |
6 | use std::sync::atomic::{AtomicI64, Ordering};
  |                         ^^^^^^^^^
  |                         |
  |                         no `AtomicI64` in `sync::atomic`
  |                         help: a similar name exists in the module: `AtomicI32`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `pyo3` (lib) due to previous error

Caused by:
  process didn't exit successfully: `rustc --crate-name pyo3 --edition=2021 /home/mgorny/.cargo/registry/src/index.crates.io-9850fa1df3136423/pyo3-0.20.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=135 --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no --cfg 'feature="default"' --cfg 'feature="extension-module"' --cfg 'feature="indoc"' --cfg 'feature="macros"' --cfg 'feature="pyo3-macros"' --cfg 'feature="unindent"' -C metadata=c3280b3684a9dc16 -C extra-filename=-c3280b3684a9dc16 --out-dir /home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps -L dependency=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps --extern cfg_if=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libcfg_if-dd66dfbd15e1f200.rmeta --extern indoc=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libindoc-330c4043caad1819.so --extern libc=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/liblibc-bd25ee420bfedf7b.rmeta --extern memoffset=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libmemoffset-80b40800efcae77e.rmeta --extern parking_lot=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libparking_lot-166554cbb51c52ff.rmeta --extern pyo3_ffi=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libpyo3_ffi-d8ceef02bf595e0b.rmeta --extern pyo3_macros=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libpyo3_macros-b6c77a4bfb6a873a.so --extern unindent=/home/mgorny/setuptools-rust/examples/html-py-ever/target/release/deps/libunindent-a0d1a359c22931b4.rmeta --cap-lints allow --cfg Py_3_6 --cfg Py_3_7 --cfg Py_3_8 --cfg Py_3_9 --cfg Py_3_10 --cfg Py_3_11 --cfg thread_local_const_init` (exit status: 1)
error: `cargo rustc --lib --message-format=json-render-diagnostics --manifest-path Cargo.toml --release -v --features pyo3/extension-module --crate-type cdylib --` failed with code 101

ERROR Backend subprocess exited when trying to invoke build_wheel
```


### Your operating system and version

Gentoo Linux, ppc32 nspawn container on POWER9

### Your Python version (`python --version`)

Python 3.11.6

### Your Rust version (`rustc --version`)

rustc 1.71.1 (eb26296b5 2023-08-03)

### Your PyO3 version

0.20.0

### How did you install python? Did you use a virtualenv?

Gentoo package. Tested inside a venv (with another venv layer via build) and outside.

### Additional Info

I know that C programs using 64-bit atomics need to explicitly link `-latomic` here. I don't know how that maps to Rust.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,6 +19,7 @@ cfg-if = "1.0"
 libc = "0.2.62"
 parking_lot = ">= 0.11, < 0.13"
 memoffset = "0.9"
+portable-atomic = "1.0"
 
 # ffi bindings to the python interpreter, split into a separate crate so they can be used independently
 pyo3-ffi = { path = "pyo3-ffi", version = "=0.21.0-dev" }

--- a/newsfragments/3619.fixed.md
+++ b/newsfragments/3619.fixed.md
@@ -1,0 +1,1 @@
+Use portable-atomic to support platforms without 64-bit atomics

--- a/src/impl_/pymodule.rs
+++ b/src/impl_/pymodule.rs
@@ -3,7 +3,7 @@
 use std::cell::UnsafeCell;
 
 #[cfg(all(not(PyPy), Py_3_9, not(all(windows, Py_LIMITED_API, not(Py_3_10)))))]
-use std::sync::atomic::{AtomicI64, Ordering};
+use portable_atomic::{AtomicI64, Ordering};
 
 #[cfg(not(PyPy))]
 use crate::exceptions::PyImportError;

